projects/step-core/src/lib/shared/utils.ts

```diff
@@ -37,3 +38,42 @@ export const setObjectFieldValue = (object: Record<string, unknown>, fieldPath:
     return (res as Record<string, unknown>)[fieldName];
   }, object);
 };
+
+/**
+ * The time complexity of a Breadth-First Search (BFS) algorithm is O(|V| + |E|),
+ * where |V| is the number of vertices and |E| is the number of edges in the graph being traversed.
+ * The reason for this is that in the worst-case scenario, BFS needs to visit all vertices and all edges.
+ */
+export const bfs = <T>({
```

Comment:
```suggestion
export const bfSearch = <T>({
```

I would prefer so have a naming that makes it clear this is a search